provider/docker_test.go

```diff
@@ -956,9 +999,10 @@ func TestDockerLoadDockerConfig(t *testing.T) {
 			},
 			expectedFrontends: map[string]*types.Frontend{
 				"frontend-Host-test-docker-localhost": {
-					Backend:        "backend-test",
-					PassHostHeader: true,
-					EntryPoints:    []string{},
+					Backend:              "backend-test",
+					PassHostHeader:       true,
+					EntryPoints:          []string{},
+					WhitelistSourceRange: []string{},
```

Comment:
Hmm Unfortunately reflect.DeepEqual distinguished from a []string{} initialized with nil and one initialized as a empty list, which took me quite a while because the tests fail with the unhelpful message:
```
expected map[string]*types.Frontend{"frontend-Host-test1-docker-localhost":(*types.Frontend)(0xc4201c49c0), "frontend-Host-test2-docker-localhost":(*types.Frontend)(0xc4201c4a20)},
     got map[string]*types.Frontend{"frontend-Host-test1-docker-localhost":(*types.Frontend)(0xc4202e8480), "frontend-Host-test2-docker-localhost":(*types.Frontend)(0xc4202e8600)}
```

I'd suppose it would be best to remove the reflect.DeepEqual shortcut and instead assert the expectations one by one, which would also lead to clearer messages in case of a failing test, but I don't think this should be part of this PR.

Given the above preconditions, it is indeed necessary to initialize WhitelistSourceRange with an empty style everywhere.